.buildkite/nightly_steps.yml

```diff
@@ -1,19 +1,23 @@
-retry:
-  automatic:
-    - exit_status: -1  # Connection to the Agent was lost
-      signal_reason: none
-      limit: 2
-    - exit_status: 255  # Timeout
-      signal_reason: none
-      limit: 2
-    - signal_reason: 2  # Flaky test
-      limit: 2
+definitions:
+  retries: &retries
+    retry:
+      automatic:
+        - exit_status: -1  # Connection to the Agent was lost
+          signal_reason: none
+          limit: 2
+        - exit_status: 255  # Timeout
+          signal_reason: none
+          limit: 2
+        - exit_status: 2  # Flaky test
+          signal_reason: none
```

Comment:
Late non-blocking question: `signal_reason: none` wasn't originally included for the flaky test retry. What changes by adding this?

Reply:
It's here for consistency.

`signal_reason` itself can be [one of these options](https://buildkite.com/docs/pipelines/command-step): 

- "*"
- none
- cancel
- agent_stop

For now it seems like retrying on "cancel" or "agent_stop" does not make sense, so retrying only when there's no reason.